### PR TITLE
3.1 update for gRPC-start tutorial and its two sample apps

### LINKED ISSUE #18104: 3.0 to 3.1 update for gRPC client and server tutorial and 2 sample apps
The following should be updated from ASP.NET Core 3.0 to ASP.NET Core 3.1:
- [Tutorial: Create a gRPC client and server in ASP.NET Core](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-3.1&branch=pr-en-us-18103&tabs=visual-studio)
- GrpcGreeter sample app
- GrpcGreeterClient sample app

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1d298a6c-8ddc-dbc8-d219-0fa7138442c2
* Version Independent ID: b96be346-958a-cfcd-d80a-c352d64223e8
* Content: [Create a .NET Core gRPC client and server in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-3.1&tabs=visual-studio)
* Content Source: [aspnetcore/tutorials/grpc/grpc-start.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/tutorials/grpc/grpc-start.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @JunTaoLuo
* Microsoft Alias: **johluo**

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal Review - Tutorial: Create a gRPC client and server in ASP.NET Core](https://review.docs.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-3.1&branch=pr-en-us-18103&tabs=visual-studio)

Fixes #18104

Updated for ASP.NET Core 3.1:
- grpc-start.md tutorial
- GrpcGreeter sample app
- GrpcGreeterClient sample app

Completing this issue is a prerequisite to related issue #16467 which includes creating a new 3.1 version samples .zip for download and redirecting to this updated tutorial.

### CODE REVIEW

**Comment by CHOWYANYAN** [comment]
How to add the apps in

**Comment by wadepickett** [comment]
Thanks @CHOWYANYAN.    Do you mean you need more information on:
1. How download and open the finished apps in Visual Studio.
or
2. How to publish on a server?
or something else?

**Comment by wadepickett** [comment]
@JamesNK, does this update look OK for 3.1?

File: aspnetcore/tutorials/grpc/grpc-start/sample/GrpcGreeter/GrpcGreeter.csproj [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+    <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
     <Protobuf Include="Protos\greet.proto" GrpcServices="Server" />
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Grpc.AspNetCore" Version="2.23.2" />
+    <PackageReference Include="Grpc.AspNetCore" Version="2.27.0" />
```
Comment (JamesNK) [inline aspnetcore/tutorials/grpc/grpc-start/sample/GrpcGreeter/GrpcGreeter.csproj:12]: 2.28.0

**Pull request merged**